### DISCUSSION TITLE
Generate  and update  permissions list from all models

### DISCUSSION THREAD

**Discussion by webapps-biz-pl** [spatie/laravel-permission#1914] [Ideas]
My suggestion is (L8):<br>
<pre><code>

namespace App\Console\Commands;

use Facade\Ignition\Support\ComposerClassMap;
use Illuminate\Console\Command;
use Illuminate\Support\Str;
use Spatie\Permission\Models\Permission;

class GeneratePermissionsList extends Command
{
    /**
     * The name and signature of the console command.
     *
     * @var string
     */
    protected $signature = 'permission:generate-permissions-list';

    /**
     * The console command description.
     *
     * @var string
     */
    protected $description = 'Generating and supplementing permissions on the basis of existing models.';

    /**
     * Create a new command instance.
     *
     * @return void
     */
    public function __construct()
    {
        parent::__construct();
    }

    /**
     * Execute the console command.
     *
     * @return int
     */
    public function handle()
    {
        $models = [];
        $permissionsList = [];
        $actions = ['list', 'view', 'create', 'update', 'delete'];
        $classMap = new ComposerClassMap();
        $listClasses = $classMap->listClasses();
        foreach ($listClasses as $key => $value) {
            if ($key == 'App\Models\Scopes\Searchable') {
                continue;
            }
            if (substr($key, 0, 11) == 'App\Models\\') {
                $models[$key] = Str::plural(
                    strtolower(
                        str_replace('App\Models\\', '', $key)
                    )
                );
            }
        }
        unset($classMap, $listClasses, $key, $value);
        if (count($models) > 0) {
            foreach ($models as $model) {
                foreach ($actions as $action) {
                    $permissionsList[] = trim($action . ' ' . $model);
                }
            }
        }
        unset($models, $action);
        foreach ($permissionsList as $value) {
            $line = "Permission: " . $value;
            if (Permission::where('guard_name', 'web')->where('name', $value)->count() < 1) {
                $p = new Permission();
                $p->name = $value;
                $p->guard_name = 'web';
                $p->save();
                $this->info($line . " - created.");
            } else {
                $this->warn($line . " - already exists.");
            }
        }

        return Command::SUCCESS;
    }
}
</code>
</pre>

**Comment by erikn69** [comment]
It doesn't look very workable, there is no rule that should specifically use `['list', 'view', 'create', 'update', 'delete']`, and some use uppercase and lowercase, subscript, and other keywords, and not even the names of the models are a rule

but as a customization option it seems fine to me

**Reply by webapps-biz-pl** [reply]
This is just my suggestion to be developed by people smarter than me :-)
I really miss this functionality, so I wrote something like this for my own needs.
